[ADD] website: create and import website (#18610)
odoo/documentation: content/applications/websites/website/website_creation.rst

================
Website creation
================

Odoo Website enables you to create and customize a website for presenting information, or connecting

> it with an :doc:`online shop <../ecommerce>`, :doc:`eLearning platform <../elearning>`, :doc:`forum

Reviewer: ```suggestion
Odoo Website enables you to create and customize a website for presenting information, or integrating an :doc:`online shop <../ecommerce>`, :doc:`eLearning platform <../elearning>`, :doc:`forum
```

> <../forum>`, or :doc:`blog <../blog>`. You can either :ref:`create a new website

Reviewer: ```suggestion
<../forum>`, and/or :doc:`blog <../blog>`. You can either :ref:`create a new website
```
(since we can integrate all of them on the same website)

<website_creation/create>` or :ref:`import an existing one <website_creation/import>`.

.. _website_creation/create:

Create a new website
====================

.. _website_creation/website-builder:

After installing the **Website** app, Odoo prompts you to use its 4-step website builder to create
the website. You can:

#. Define the purpose of your website.
#. :guilabel:`Choose a Color Palette` or :icon:`fa-cloud-upload` :guilabel:`Upload` your logo so

>    Odoo can detect a matching color scheme and generate a brand style.

Reviewer: ```suggestion
#. :guilabel:`Choose a Color Palette` or :icon:`fa-cloud-upload` :guilabel:`Upload` a logo to
   detect a matching color scheme and generate a brand style accordingly.
```

   </applications/marketing/events>` page.
#. :guilabel:`Choose your favorite Theme`. You can still change and :doc:`customize
   <web_design/themes>` it later.

If you prefer to bypass this setup, click the :guilabel:`Skip and start from scratch` link.

.. note::
   Starting from scratch results in a blank website without a predefined design or content.

To create an additional website, follow these steps:

#. Navigate to :menuselection:`Website --> Configuration --> Settings`.
#. In the settings banner, click :icon:`fa-plus` :guilabel:`New Website`.
#. In the :guilabel:`Add Website` pop-up window:

   - Enter a :guilabel:`Website Name`.
   - Fill in the :doc:`Website Domain <configuration/domain_names>`, if applicable.
   - In a :doc:`multi-company environment </applications/general/companies/multi_company>`, select
     a :guilabel:`Company`.
   - In a :doc:`multilingual environment <configuration/translate>`, select :guilabel:`Languages`,
     and define a :guilabel:`Default Language`.
   - If applicable, define the number of :ref:`karma points <forum/karma>` required to view other

>      users' profiles (150 by default).

Reviewer: this seems so, so niche, I don't know if we should even mention it here? 🙈 it seems so minor compared to the other fields/options, especially since the field is filled by default and we could probably change this later if needed?

Author: Hmm I get what you mean... I can remove it. Just thought to make it the most complete.

#. Click :guilabel:`Create`.
#. Use the :ref:`website builder <website_creation/website-builder>` to complete the setup.

.. _website_creation/import:

Import an existing website
==========================

The website import tool works like an OCR scan: it analyzes an existing website and recreates it in

> Odoo by fetching its pages and rebuilding them as Odoo website pages. Images, the sitemap, the

Reviewer: - I'd use an abbreviation for OCR and I wonder if it just **works like** an OCR "scan" or does it actually **use** OCR to "scan" the website (see the wording of [this doc page's](https://www.odoo.com/documentation/master/applications/finance/accounting/vendor_bills/invoice_digitization.html) intro paragraph for reference for where my comment is coming from)

- Maybe, just like in the linked doc page I just mentionned above, it would be useful to briefly mention the tools it uses to analyze the existing website and recreate it in odoo, would that make sense here?

- And then I'd simplify the sentence slightly I think (+ add the info, if you choose to, from my comment above)

```suggestion
The website import tool uses a web scraper to analyze an existing website. It then uses custom AI models to fetch the original pages and rebuilds them as Odoo website pages. Images, the sitemap, the
```

Author: I think it works like an OCR, but does not use it. I think it uses a web scraper. I will discuss with Christopher

EDIT: I confirm! We use a web scraper that analyzes the web pages and splits them into sections. It then matches the sections with Odoo snippets. Then visuals are extracted and also transferred to those snippets. Odoos own AI models help with that.

Reviewer: EDIT to original comment: disregard 2 first list items in my original comment, just keep the suggestion (fact checked)

:ref:`header <website/header_footer/header-design>`, and the :ref:`footer
<website/header_footer/footer-design>` are also imported. Optionally, :doc:`products

> <../ecommerce/configuration/products>` can be imported at the same time. The tool converts pages

Reviewer: see my [comment a bit further down](https://github.com/odoo/documentation/pull/18610/changes#r3520014966) about product import

into standard Odoo :doc:`building blocks <web_design/building_blocks>`, making the imported website

> fully editable in Odoo.

Reviewer 2: Might be worth mentioning somewhere that this tool is only available on enterprise

.. note::
   The imported website is not an exact copy or pixel-perfect clone, and some manual adjustments may

>    be needed after import.

Reviewer: ```suggestion
   The imported website is not an exact replica of the original. Manual adjustments to layout, styling, or content may be needed after import.
```

.. important::
   - By using this tool, you confirm that you own the website and all its associated content (such
     as images), and that you take full responsibility for their use and any potential legal
     implications.
   - For the best results, ensure your database is up to date with the latest version of Odoo.
   - Do not edit the website during the import process. Importing data overwrites the content of
     the website currently being edited, potentially resulting in the loss of unsaved changes.

> To import your current website, follow these steps:

Reviewer: Is any configuration necessary prior to the steps outlined below? 
When I tried to test this, I repeatedly had this error message:

<img width="426" height="152" alt="Image" src="https://github.com/user-attachments/assets/4e343852-fc72-49d1-aae4-6ac7f4ae32c1" />

or is that related to the enterprise thing? or is it linked to the original website's platform (i think the website i used uses shopify but i'm not sure)

and then I tried using the trial import tool and I had this:

<img width="410" height="113" alt="Image" src="https://github.com/user-attachments/assets/4b470066-7a51-4ad9-8daa-d2bf9d3df06f" />

does that mean we need IAP credits, just [like for document digitization](https://www.odoo.com/documentation/master/applications/finance/accounting/vendor_bills/invoice_digitization.html#pricing)?

Reviewer 2: Which version of Odoo did you test this on? 

The issue `API Check failed.` is probably because you click submit before the tool could figure out which APIs were available for the website you tried to import. If you could let me know which version you used, I can try reproduce the issue and fix it.

The issue `Not enough credits for data extraction.` is not related to us. Were you using runbot? My guess is that someone else was also on the runbot db trying the OCR or something which requires credits. The scraper does not require any credits.

Edit: I tried on runbot on saas-19.3 and don't seem to get the issues you get.

#. Navigate to :menuselection:`Website --> Configuration --> Settings`.
#. In the settings banner, click :icon:`fa-plus` :guilabel:`Import Website`.

> #. On the :guilabel:`Import your Pages and Products` page, enter the :guilabel:`Website URL` (e.g.,

Reviewer: just a quick note (i'm sure you checked already or were going to for the fw-ports, but just in case!) that in more recent versions (i tested on 19.3 but it might even be earlier?), the page has become a popup, with more options, and Import products is not enabled by default

Reviewer 2: Yes from 19.1 it becomes a popup. So for the 19.1 version it will be noted :)

Edit: FYI, I made a document for LMAN to explain all the changes between the versions from 19.0 to 19.3. If you would like access, you can ping me on discord

   `https://www.example.com`).
#. When running an online shop, ensure the :guilabel:`Import Products` option is enabled. By

>    default, all detected products are imported.

Reviewer: When I tested this (with several different websites, and in 19.3 which I assume has an updated version of the import tool), the products were not imported into the db? is that always the case or just part of the "the imported website is not a perfect copy" thing? 🙈  

We might want to rephrase this and some parts of the note below as well if it is indeed always the case

Reviewer 2: Could you provide the URLs you tested? Did they contain products?

If the products are not supported by the tool then it will not work. However, if the products are supported then it should work very well. For example, if the website was a Shopify website and they have products, it should extract the products without any issue.

IMO, the 'detected' implies that it has to be able to see the products to import them.

Reviewer: What do you mean exactly by *if the products are not supported by the tool*? which products are supported and which aren't? 
I tested these websites: https://www.edgardcooper.com/ and https://myfleeters.com/

Reviewer: EDIT: I think I get it now with your other reply; we mean products from a supported platform, right? @lman-odoo I think I would mention this here as well, very briefly?

Reviewer 2: Thanks for the websites! My Fleeters is a Shopify website. I tried with @lman-odoo and it seemed to work so I'm not sure what happened with your case. I checked the server and it should have worked. If you have more info, please let me know.

> products are not supported by the tool

This means that the website is not hosted on any of the platforms we support (Shopify, WooCommerce, PrestaShop) and there is no metadata information on the products.

On the contrary, the products are supported, or better said "the products can be extracted" if the website is a Shopify/WooCommerce/PrestaShop website or it has at least some metadata on the products.

Reviewer: Thank you so much for checking and for taking the time to clarify this once again, I really appreciate it :hugs: 
And I like the "products can be extracted" phrasing, I think that's a bit more accurate and clear, too!

and yeah i don't know why it didn't work for me, but it's fine, i'm glad you could check, thank you!

#. To submit the request, click :guilabel:`Import my website`.

> Once the import is complete, you will receive a notification with details about the number of pages

Reviewer: ```suggestion
Once the import is complete, you will receive an email with details about the number of pages
```
or am I confusing this with the email we receive when the website's ready? (but then I didn't see any other notification? 🙈 )

If the email and notification are the same thing, I don't see any details in that email? just a generic message and a Go to website button

Reviewer 2: It depends on the DB settings.By default it's an email but it could a notification in Odoo.

<img width="1120" height="720" alt="image" src="https://github.com/user-attachments/assets/a1e81848-2540-488c-b34a-f1867777ccd9" />

For the details, that is a good point. The details are added to the email from version 19.1, so for this 19.0 PR we can remove the 'details' part.

and products imported. You can then further customize the website using the :doc:`website editor
<web_design>`.

.. note::

>    - Blogs cannot be imported.

Reviewer: just a heads up that in more recent versions they can 🙂

   - You can import up to **100** pages by default. Contact `Odoo support

>      <https://www.odoo.com/help>`_ if you need more.

Reviewer 2: ```suggestion
   - The maximum number of pages you can import is 100.
```

Since the 100 pages is hardcoded for 19.0 (due to lack of infrastructure at that time to support more), we cannot offer the option to give more pages.

I also made the fix on enterprise yesterday: http://github.com/odoo/enterprise/pull/121669

Author: Okay, thank you 😄

>    - It is not possible to import products without importing the website.

Reviewer: well, technically speaking, and unrelated to the website app, people *can* [import products into their db](https://www.odoo.com/documentation/master/applications/essentials/export_import_data.html#import-data-into-odoo) without importing a website. It might be worth rephrasing to be a bit more precise here 🙂

Reviewer 2: True - for the further versions (19.1+), it might be also worth linking that documentation and this documentation since you can use the tool to automatically import all your products if the platform is supported. Would be much much easier for the user than trying to figure it out with the csv files.

Reviewer: probably, but if they're limited to 20 products, they will probably need another solution for the rest anyways, right?
But yes @lman-odoo a tip/note in the import doc would be nice I think.

and @chrisdt1998 I think we should maybe provide a list of supported platforms, or non-supported platforms, depending on how many are supported, maybe? just to make sure we're not misleading people by saying something works and then it doesn't for their specific website

Reviewer 2: True, for 19.0 they are limited to 20 products but from saas-19.1 and newer, we can import all products (well technically up to 100 000 but I haven't seen a website over 30 000 yet).

Yes, it already says in the note that we support Shopify products and any products that have metadata in the format of JSON-LD, RDFa, Open Graph or Microdata. 

In future versions, we also support WooCommerce and PrestaShop.

Reviewer: Ah yes, ok, thank you so much for clarifying! @lman-odoo I think we should maybe rephrase the note slightly to something like "Only products from websites built with Shopify or using one of the following metadata schemes are supported: (followed by the list of schemes)"

   - The import feature supports up to **20** products.
   - Importing products automatically installs the :doc:`eCommerce app <../ecommerce>` if it is not
     already installed.

>    - You can import products from `Shopify <https://www.shopify.com/>`_ or by using supported

Reviewer: - How exactly? (I don't necessarily think we need to explain the details, but maybe a general idea would be helpful). 
- I assume the 20-product limit still applies here?

Based on the answer(s), it might be worth it to rephrase this slightly 🙂

Author: The product extraction is done using an public API. In this case for Shopify, but you don't need any API key as it is public.
And we can read metadata for that.
And yes, we can only import 20 products.

Reviewer: See my [comment above](https://github.com/odoo/documentation/pull/18610/changes#r3535180020) about rephrasing the note

     metadata schemas. The following schemas are supported:

     - JSON-LD
     - RDFa
     - Open Graph
     - Microdata

.. tip::

>    - You can use the `Website Import Tool Trial <https://www.odoo.com/fr_FR/page/startwebsite>`_

Reviewer: I think we could simplify the wording slightly, and include that this actually creates a trial db:

```suggestion
   - Use the `Website Import Tool trial <https://www.odoo.com/page/startwebsite>`_ to test the website import feature by importing up to 10 pages from the original website into a trial database.
```

     to import a sample of up to 10 pages, which is ideal for testing.

>    - Ensure your website is publicly accessible and not protected by :ref:`Cloudflare

Reviewer: ```suggestion
   - Ensure the original website is publicly accessible and not protected by :ref:`Cloudflare
```
(otherwise we might confuse it with the odoo website)

the website. You can:

#. Define the purpose of your website.
#. :guilabel:`Choose a Color Palette` or :icon:`fa-cloud-upload` :guilabel:`Upload` a logo to detect
   a matching color scheme and generate a brand style accordingly.
#. :guilabel:`Add Pages and Features`, such as an :doc:`online shop <../ecommerce>` or :doc:`events
   </applications/marketing/events>` page.
#. :guilabel:`Choose your favorite Theme`. You can still change and :doc:`customize
   <web_design/themes>` it later.

If you prefer to bypass this setup, click the :guilabel:`Skip and start from scratch` link.

.. note::
   Starting from scratch results in a blank website without a predefined design or content.

To create an additional website, follow these steps:

#. Navigate to :menuselection:`Website --> Configuration --> Settings`.
#. In the settings banner, click :icon:`fa-plus` :guilabel:`New Website`.
#. In the :guilabel:`Add Website` pop-up window:

   - Enter a :guilabel:`Website Name`.
   - Fill in the :doc:`Website Domain <configuration/domain_names>`, if applicable.
   - In a :doc:`multi-company environment </applications/general/companies/multi_company>`, select
     a :guilabel:`Company`.

>    - In a :doc:`multilingual environment <configuration/translate>`, select the relevant

Reviewer: I'd make this a numbered sub-list i think since these are steps?

Author: Would you start the sublist with 1., 2. etc. or rather a., b. etc. to not confuse?

<img width="822" height="459" alt="Image" src="https://github.com/user-attachments/assets/0a9232b0-d028-4a40-9b86-b69921431b0f" />
or

<img width="812" height="500" alt="Image" src="https://github.com/user-attachments/assets/cafe8419-d90e-4a93-b588-8c7798152e96" />

     :guilabel:`Languages` and define a :guilabel:`Default Language`.

     .. note::
        Languages already installed in the database are selected automatically.

#. Click :guilabel:`Create`.
#. Use the :ref:`website builder <website_creation/website-builder>` to complete the setup.

.. _website_creation/import:

Import an existing website
==========================

.. important::
   The website import tool is available only with an `Odoo Enterprise subscription

>    <https://www.odoo.com/pricing>`_.

Reviewer: I think I'd send people to this link instead, since "your" link doesn't say anything about community vs enterprise?: https://www.odoo.com/page/editions

The website import tool is a website scraper that searches for the website you want to import and

> analyzes it. Then, it uses custom AI models to rebuild it as an Odoo website. The extraction is

Reviewer: ```suggestion
that searches for the specified website and analyzes it. It then uses custom AI models to rebuild it as an Odoo website. The extraction is
```

#. Navigate to :menuselection:`Website --> Configuration --> Settings`.
#. In the settings banner, click :icon:`fa-plus` :guilabel:`Import Website`.
#. On the :guilabel:`Import your Pages and Products` page, enter the :guilabel:`Website URL` (e.g.,
   `https://www.example.com`).
#. When running an online shop, ensure the :guilabel:`Import Products` option is enabled to extract
   product data.
#. To submit the request, click :guilabel:`Import my website`.

Once the import is complete, you will receive an email or a notification confirming that your Odoo
website is ready. You can then further customize the website using the :doc:`website
editor <web_design>`.

.. note::
   - Blogs cannot be imported.
   - You can import up to **100** pages by default.
   - It is **not** possible to import products using the website import tool *without* importing the
     website.
   - The tool supports importing up to **20** products. For larger product catalogs, :doc:`use a
     .csv file </applications/sales/sales/products_prices/products/import>`.
   - Importing products automatically installs the :doc:`eCommerce app <../ecommerce>` if it is not

>      already installed.

Reviewer: probably a nitpick but I think i'd move this list item about the ecommerce app under the 1st list item about the 100 page limitation, before we mention all of the other ecommerce-related limitations?
